go/batch-submitter/txmgr/txmgr.go

```diff
@@ -229,14 +238,24 @@ func WaitMined(
 
 	for {
 		receipt, err := backend.TransactionReceipt(ctx, txHash)
-		if receipt != nil {
-			return receipt, nil
-		}
+		switch {
+		case receipt != nil:
+			txHeight := receipt.BlockNumber.Uint64()
+			tipHeight, err := backend.BlockNumber(ctx)
+			if err != nil {
+				log.Error("Unable to fetch block number", "err", err)
+				break
+			}
 
-		if err != nil {
+			if txHeight+numConfirmations <= tipHeight+1 {
+				return receipt, nil
+			}
+
```

Comment:
It might be useful to have a trace logline in the `case receipt != nil` with the values used in the `if` branch

Reply:
done!